torch/csrc/jit/mobile/function.cpp

```diff
@@ -1,25 +1,42 @@
 #include "function.h"
 #include "interpreter.h"
+#include <torch/csrc/jit/instruction.h>
+#include <ATen/core/op_registration/op_registration.h>
 
 namespace torch{
 namespace jit{
+char const * toString(OpCode op);
 namespace mobile {
 Function::Function(c10::QualifiedName name)
     : name_(name), code_(std::make_shared<Code>()) {}
 
 void Function::append_instruction(OpCode op, int N, int X) {
+  TORCH_CHECK(isOpSupportedInMobile(op), toString(op),
+              " is not supported in mobile module.");
   code_->instructions_.emplace_back(op, N, X);
 }
 
 void Function::append_operator(const std::string& name,
-                             const std::string& overload_name) {
+                               const std::string& overload_name) {
+  // Keep the original opname in code_
   code_->op_names_.emplace_back(name, overload_name);
+
+  //
   auto opname = code_->op_names_.back();
+  // Special treatments for some prim ops.
```

Comment:
This function should probably just not be called for OPN instructions. A separate append_vararg_operator should be called that does the resolution into a separate vararg table.